draft-ietf-jsonpath-base.md

```diff
@@ -1501,8 +1505,7 @@ immediate contexts, which lead to the following conditions for well-typedness:
 
 {:vspace}
 As a `test-expr` in a logical expression:
-: The function's declared result type is `LogicalType`, or
-  (giving rise to conversion as per {{type-conv}}) `NodesType`.
+: The function's declared result type is `LogicalType`.
```

Comment:
This change would mean a function with declared type NodesType is no longer well-typed here.
I think Glyn is right that this section needs to be consistent on its own.